File: 0812-rotate-string/0812-rotate-string.cpp
class Solution {
public:
    bool rotateString(string s, string goal) {
       if((int)s.size() != (int)goal.size()) return false ; 
       for(int i = 0  ; i < (int)goal.size() ; ++i){
            shift(s) ; 
            if(s == goal) return true ;  
            cout<< s<<'\n' ; 
       } 
       return false ; 
    }
private:
    void shift(string &str){
            char ch = str[0] ; 
            str.push_back(ch) ; 
        for(int i =1  ; i <= (int)str.size() ; ++i){
            str[i-1] = str[i] ; 
        }
        str.pop_back() ; 
    }
};

File: 0812-rotate-string/0812-rotate-string.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        maxShifts = len(s)
        print(maxShifts)
        newString = s 
        for i in range(maxShifts) :
            newString = self.shiftString(newString)
            if newString == goal :
                return True 
        return False
    def shiftString(self , s: str) -> str :
        s += s[0]
        s = s[1:]
        return s